src/main/java/gov/usgs/owi/nldi/controllers/RedirectController.java

```diff
@@ -0,0 +1,42 @@
+package gov.usgs.owi.nldi.controllers;
+
+import gov.usgs.owi.nldi.dao.LookupDao;
+import gov.usgs.owi.nldi.dao.StreamingDao;
+import gov.usgs.owi.nldi.services.ConfigurationService;
+import gov.usgs.owi.nldi.services.LogService;
+import gov.usgs.owi.nldi.services.Navigation;
+import gov.usgs.owi.nldi.services.Parameters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+
+import io.swagger.v3.oas.annotations.Hidden;
+
+@RestController
+public class RedirectController extends BaseController {
+
+	private String urlRoot = null;
+
+	public RedirectController(LookupDao inLookupDao, StreamingDao inStreamingDao, Navigation inNavigation, Parameters inParameters, ConfigurationService inConfigurationService, LogService inLogService) {
+		super(inLookupDao, inStreamingDao, inNavigation, inParameters, inConfigurationService, inLogService);
+	}
+
+
+	//Used for integration test ... see RedirectControllerIT.java
+	void setRootUrl(String url) {
```

Comment:
This is a very ugly solution, but the only one I could come up with.  Unlike WQP-WQX-Services, in nldi-services there is a TestConfigurationService where the host is hardcoded to "owi-test.usgs.gov", and other tests seem to rely on this.

So without this fix, the swagger test will always fail because it tries to redirect to owi-test.usgs.gov/nldi/swagger, etc.

If there is a better approach, that would be great.

Reply:
The fix for this was to upgrade to JUnit 5, then the RedirectTest from WQP-WQX-Services could work.